src/workflow/ArcanistLandWorkflow.php

```diff
@@ -267,12 +268,20 @@ public function run() {
     $this->readArguments();
 
     $engine = null;
+    $uberShadowEngine = null;
     if ($this->isGit && !$this->isGitSvn) {
       $engine = new ArcanistGitLandEngine();
       if ($this->shouldUseSubmitQueue && !$this->tbr) {
-        $engine = new UberArcanistSubmitQueueEngine(
-          $this->submitQueueClient,
-          $this->getConduit());
+        // If the shadow-mode is on, then initialize the shadowEngine
```

Comment:
Your initialization of `$uberShadowEngine` and `$engine` is the same here.
If initialization is the same, it is enough to set boolean value:
```
if ($this->submitQueueShadowMode) {
  $uberShadowEngineEnabled = true;
}
...
```